src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -828,6 +879,185 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
             }
         }
 
+        public async Task WaitForDatabaseNotificationOnCluster(string databaseName, long index,
+            Dictionary<string, string> clusterNodes, // (nodeTag, url)
+            HashSet<string> databaseNodes, // nodeTags
+            CancellationToken token)
+        {
+            try
+            {
+
+                foreach (var (tag, nodeUrl) in clusterNodes)
+                {
+                    if (tag == ServerStore.NodeTag)
+                    {
+                        await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, index);
+
+                        if (databaseNodes.Contains(tag))
+                        {
+                            try
+                            {
+                                var database = await ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName);
+                                if (database != null) // database isn't deleted already from current node 
+                                {
+                                    await database.RachisLogIndexNotifications.WaitForIndexNotification(index, token);
+                                }
+                            }
+                            catch (Exception ex) when (ex is DatabaseDoesNotExistException or
+                                                           DatabaseNotRelevantException or
+                                                           DatabaseDisabledException)
+                            {
+                            }
+                        }
+
+                    }
+                    else
+                    {
+                        try
+                        {
+                            using (var requestExecutor =
```

Comment:
let's create it once per server.
and sent a list of databases we want to wait for.

Reply:
Done
I created the request executor once for every server and reused it